src/sqlancer/mysql/ast/MySQLCaseOperator.java

```diff
@@ -0,0 +1,48 @@
+package sqlancer.mysql.ast;
+
+import java.util.List;
+
+import sqlancer.common.ast.newast.NewCaseOperatorNode;
+
+public class MySQLCaseOperator extends NewCaseOperatorNode<MySQLExpression> implements MySQLExpression {
+
+    public MySQLCaseOperator(MySQLExpression switchCondition, List<MySQLExpression> whenExprs,
+            List<MySQLExpression> thenExprs, MySQLExpression elseExpr) {
+        super(switchCondition, whenExprs, thenExprs, elseExpr);
+    }
+
+    @Override
+    public MySQLConstant getExpectedValue() {
+        int nrConditions = getConditions().size();
```

Comment:
It's great that you made the effort to implement the PQS component as well, but I think it would also be fine, in principle, to just bail out (e.g., throw an `IgnoreMeException` or so).

Reply:
But if this works, it would be great to also merge this part.

Reply:
Sure, thanks! I have added some tests to validate the `getExpectedValue` method and added a screenshot of a SQLancer run in the PR description.